app/models/audit_trail.rb

```diff
@@ -71,4 +32,33 @@ def changed_by_user
   def marked_urgent?
     modified.include?('urgent_flag') && modified['urgent_flag'] == true
   end
+
+  def shaped_changes
+    orig = original.reject { |field| AuditTrail::IRRELEVANT_FIELDS.include? field }
+    mod = modified.reject { |field| AuditTrail::IRRELEVANT_FIELDS.include? field }
+    all_fields = orig.keys | mod.keys
+    changeset = {}
+    all_fields.each do |key|
+      changeset[key.humanize] = { original: format_fieldchange(key, orig[key]),
+                                  modified: format_fieldchange(key, modified[key]) }
+    end
+    changeset
+  end
+
+  private
+
+  def format_fieldchange(key, value)
```

Comment:
This rolls up the shaping functions. I don't like that L51 and L61 do the same thing, but the early return on L51 prevents us from making a DB call on clinic that will fail. Is this sloppy on my part? Possibly! Does it make the app error? nah!